sdk/tables/azure-data-tables/samples/async_samples/sample_conditional_update_async.py

```diff
@@ -0,0 +1,115 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_conditional_update_async.py
+
+DESCRIPTION:
+    These samples demonstrate how to update Tables storage table conditionally. The way to update Tables cosmos table is similar.
+
+USAGE:
+    python sample_conditional_update_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) TABLES_STORAGE_ENDPOINT_SUFFIX - the Table storage service account URL suffix
+    2) TABLES_STORAGE_ACCOUNT_NAME - the Tables storage account name
+    3) TABLES_PRIMARY_STORAGE_ACCOUNT_KEY - the Tables storage account access key
+"""
+import asyncio
+import os
+from datetime import datetime
+from dotenv import find_dotenv, load_dotenv
+from uuid import uuid4
+from azure.data.tables.aio import TableClient
+from azure.data.tables._models import UpdateMode
+from azure.core import MatchConditions
+from azure.core.exceptions import ResourceExistsError, ResourceModifiedError
+
+class ConditionalUpdateSamples(object):
+    def __init__(self):
+        load_dotenv(find_dotenv())
+        self.access_key = os.getenv("TABLES_PRIMARY_STORAGE_ACCOUNT_KEY")
+        self.endpoint_suffix = os.getenv("TABLES_STORAGE_ENDPOINT_SUFFIX")
+        self.account_name = os.getenv("TABLES_STORAGE_ACCOUNT_NAME")
+        self.endpoint = "{}.table.{}".format(self.account_name, self.endpoint_suffix)
+        self.connection_string = (
+            "DefaultEndpointsProtocol=https;AccountName={};AccountKey={};EndpointSuffix={}".format(
+                self.account_name, self.access_key, self.endpoint_suffix
+            )
+        )
+        self.table_name_prefix = "SampleConditionalUpdate"
+        self.entity1 = {
+            "PartitionKey": "color",
+            "RowKey": "brand",
+            "text": "Marker",
+            "color": "Purple",
+            "price": 4.99,
+            "last_updated": datetime.today(),
+            "product_id": uuid4()
+        }
+        self.entity2 = {
+            "PartitionKey": "color",
+            "RowKey": "brand",
+            "color": "Red",
+            "inventory_count": 42,
+            "barcode": b"135aefg8oj0ld58" # cspell:disable-line
+        }
+
+    async def conditional_update_basic(self):
+        async with TableClient.from_connection_string(self.connection_string, self.table_name_prefix + uuid4().hex) as table_client:
+            await table_client.create_table()
+            metadata1 = await table_client.create_entity(entity=self.entity1)
+            print("Entity:")
+            print(self.entity1)
+
+            # Merge properties of an entity with one that already existed in a table
```

Comment:
Maybe add something like: `This operation will only succeed if the entity has not been modified since we last retrieved the Etag.`